recipes/spdlog/all/conanfile.py

```diff
@@ -102,15 +104,23 @@ def package_id(self):
 
     def package_info(self):
         if self.options.header_only:
-            self.cpp_info.defines = ["SPDLOG_FMT_EXTERNAL"]
+            self.cpp_info.components["spdlog_header_only"].defines.append("SPDLOG_FMT_EXTERNAL")
+            if self.options.wchar_support:
+                self.cpp_info.components["spdlog_header_only"].defines.append("SPDLOG_WCHAR_TO_UTF8_SUPPORT")
+            if self.options.wchar_filenames:
+                self.cpp_info.components["spdlog_header_only"].defines.append("SPDLOG_WCHAR_FILENAMES")
+            if self.options.no_exceptions:
+                self.cpp_info.components["spdlog_header_only"].defines.append("SPDLOG_NO_EXCEPTIONS")
+            if self.settings.os == "Linux":
+                self.cpp_info.components["spdlog_header_only"].system_libs = ["pthread"]
         else:
             self.cpp_info.libs = tools.collect_libs(self)
             self.cpp_info.defines = ["SPDLOG_COMPILED_LIB", "SPDLOG_FMT_EXTERNAL"]
-        if self.options.wchar_support:
-            self.cpp_info.defines.append("SPDLOG_WCHAR_TO_UTF8_SUPPORT")
-        if self.options.wchar_filenames:
-            self.cpp_info.defines.append("SPDLOG_WCHAR_FILENAMES")
-        if self.options.no_exceptions:
-            self.cpp_info.defines.append("SPDLOG_NO_EXCEPTIONS")
-        if self.settings.os == "Linux":
-            self.cpp_info.system_libs = ["pthread"]
+            if self.options.wchar_support:
+                self.cpp_info.defines.append("SPDLOG_WCHAR_TO_UTF8_SUPPORT")
+            if self.options.wchar_filenames:
+                self.cpp_info.defines.append("SPDLOG_WCHAR_FILENAMES")
+            if self.options.no_exceptions:
+                self.cpp_info.defines.append("SPDLOG_NO_EXCEPTIONS")
+            if self.settings.os == "Linux":
+                self.cpp_info.system_libs = ["pthread"]
```

Comment:
Open to suggestions

Reply:
Maybe always create a component?
```python
if self.options.header_only:
    component_name = "spdlog_header_only"
    self.cpp_info.components[component_name].defines.append("SPDLOG_FMT_EXTERNAL")
else:
    component_name = "libspdlog"
    self.cpp_info.components[component_name].libs =  tools.collect_libs()
if self.options.no_exceptions:
    self.cpp_info.components[component_name].defines.append("SPDLOG_NO_EXCEPTIONS")
if self.settings.os == "Linux":
    self.cpp_info.components[component_name].system_libs = ["pthread"]
...
```

As long as you use at least once `self.cpp_info.components[component_name].XXX`

Reply:
The second compile one, would it not generate `spdlog::libspdlog` as a target? Is that okay since there's always a `spdlog::spdlog` with _all_ components (ie just one) which is what is required?

Reply:
> Open to suggestions

I thought your question was about pthreads... See #1849

Reply:
> The second compile one, would it not generate `spdlog::libspdlog` as a target? Is that okay since there's always a `spdlog::spdlog` with _all_ components (ie just one) which is what is required?

Yes, I recently learned that you can also use `"ALL"` as a component name. See https://github.com/conan-io/conan-center-index/pull/2173#issuecomment-656278330

Reply:
> The second compile one, would it not generate `spdlog::libspdlog` as a target? Is that okay since there's always a `spdlog::spdlog` with _all_ components (ie just one) which is what is required?

Yes, `spdlog::libspdlog` target will be generated. But I thought your question was about code simplicity.

Reply:
oh it was, it just the target is not the correct name, and i loved ur suggestion